ocean/src/clans.rs

```diff
@@ -1,39 +1,57 @@
-
+use std::cell::RefCell;
+use std::collections::HashMap;
 #[derive(Debug)]
 pub struct ClanSystem {
-    // TODO: add necessary fields
+    clans: RefCell<HashMap<String, Vec<String>>>,
```

Comment:
Nice choice of HashMap (instead of Vec) - makes accessing clans cheaper
Kinda interesting that you had to use RefCell here since the requirement was that each Beach has a ClanSystem (you can make the Beach own it). But this is not necessarily wrong, as long as all the functions return the correct results